reactioncommerce/reaction
# 3866 willopez remove collection hooks in search mongo package

Resolves #3866 
Impact: Major
Type: Refactor

## Issue
Remove Products and Orders collection hooks in mongo-search package, this is part of the effort to remove the collection hooks meteor package.

## Solution
Refactor collection hooks into reaction hooks, and call where necessary.

## Testing Instructions

**After Order Insert Hook** - Inserts order search record

All Hooks Related to mongo search package: `imports/plugins/included/search-mongo/server/hooks/search.js`

**Testing Procedures:**

Hook Name: `afterOrderInsert` - perform a project search to find call sites

User story 1: Place or order

1. Place on order
2. Verify searching for the purchase product returns a result, a new record should have been added to the `OrderSearch` collection
 
___
Hook Name: `afterUpdateOrderUpdateSearchRecord` - perform a project search to find call sites

User story 1: Start processing order

1. Place on order
2. Start processing order, capture payment.
3. Verify search still works,
___ 
Hook Name: `afterRemoveProduct` - perform a project search to find call sites

User story 1: Add a new product and then remove it
1. Create a new product
2. Verify searching for newly added product returns a result
3. Remove product, archive and then publish
4. Verify searching for the deleted product does not return a result.

___
Hook Name: `afterUpdateCatalogProduct` -  perform a project search to find call sites related to searhing

User Story 1: Search for product after updating title
1. Search for an existing product by title
2. Update product title to something different
3. Publish product
4. Verify product shows up in results by searching with the updated title.

___
Hook Name: `afterInsertProduct` - perform a project search to find call sites related to searching

User Story 1: Create a new product and search for by title
1. Create new product
2. Search by new product's title
3. Confirm new product shows up in search results.
